File: RxTools-library/src/main/java/com/vondear/rxtools/view/progressing/animation/interpolator/PathInterpolatorCompatBase.java
package com.vondear.rxtools.view.progressing.animation.interpolator;

import android.graphics.Path;
import android.view.animation.Interpolator;

/**
 * @author vondear
 * Base implementation for path interpolator compatibility.
 */
class PathInterpolatorCompatBase {

    private PathInterpolatorCompatBase() {
        // prevent instantiation
    }

    public static Interpolator create(Path path) {
        return new PathInterpolatorDonut(path);
    }

    public static Interpolator create(float controlX, float controlY) {
        return new PathInterpolatorDonut(controlX, controlY);
    }

    public static Interpolator create(float controlX1, float controlY1,
                                      float controlX2, float controlY2) {
        return new PathInterpolatorDonut(controlX1, controlY1, controlX2, controlY2);
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/loadingview/animation/interpolator/PathInterpolatorCompatBase.kt
package com.tamsiree.rxui.view.loadingview.animation.interpolator

import android.graphics.Path
import android.view.animation.Interpolator

/**
 * @author tamsiree
 * Base implementation for path interpolator compatibility.
 */
internal object PathInterpolatorCompatBase {
    fun create(path: Path?): Interpolator {
        return PathInterpolatorDonut(path)
    }

    fun create(controlX: Float, controlY: Float): Interpolator {
        return PathInterpolatorDonut(controlX, controlY)
    }

    fun create(controlX1: Float, controlY1: Float,
               controlX2: Float, controlY2: Float): Interpolator {
        return PathInterpolatorDonut(controlX1, controlY1, controlX2, controlY2)
    }
}